### PR TITLE
dotnet: suppress null warning

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

**Comment by matklad** [comment]
@batiati this was caught by github: 

https://github.com/tigerbeetle/tigerbeetle/pull/1463/files#annotation_17317769065

is there some easy way to make our build outright fail if there are such warnings?

**Review by batiati — CHANGES_REQUESTED** [review]
> is there some easy way to make our build outright fail if there are such warnings?

We just need to add `<TreatWarningsAsErrors>true</TreatWarningsAsErrors>` in the `TigerBeetle.Tests.csproj`

File: src/clients/dotnet/TigerBeetle.Benchmarks/TigerBeetle.Benchmarks.csproj [hunk 1]
```diff
@@ -5,6 +5,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <UseCurrentRuntimeIdentifier>true</UseCurrentRuntimeIdentifier>
     <RollForward>LatestMajor</RollForward>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```
Comment (batiati) [inline src/clients/dotnet/TigerBeetle.Benchmarks/TigerBeetle.Benchmarks.csproj:8]:
```suggestion
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    <Nullable>enable</Nullable>    
```

I think we can also remove `<AllowUnsafeBlocks>true</AllowUnsafeBlocks>` for tests and benchmarks; only the main library needs it set as `true`.

Btw, it's another yak asking to be shaved: the clients' benchmark projects 🦬

Comment (batiati) [reply src/clients/dotnet/TigerBeetle.Benchmarks/TigerBeetle.Benchmarks.csproj:8]: > I think we can also remove `<AllowUnsafeBlocks>true</AllowUnsafeBlocks>` for tests and benchmarks; only the main library needs it set as `true`.

No 😆, the test project needs unsafe code for testing the packet logic.
But we can remove it from the benchmark project for sure.

Comment (matklad) [reply src/clients/dotnet/TigerBeetle.Benchmarks/TigerBeetle.Benchmarks.csproj:8]: It was worth trying!

**Review by batiati — APPROVED** [review]

**Pull request merged**